projects/graphql-client/src/graphql-request.service.test.ts

```diff
@@ -90,159 +154,143 @@ describe('GraphQl Request Service', () => {
     spectator = createService();
   });
 
-  test('fires immediate queries without delay', () => {
-    spectator.service.registerHandler(buildQueryHandler());
+  test('fires isolated queries without delay', () => {
+    spectator.service.registerHandler(buildQueryHandlerOne());
     const queryFnMock = mockQueryMethod();
-    spectator.service.queryImmediately(buildRequest());
+    spectator.service.query(buildRequestOne(), { cacheability: GraphQlRequestCacheability.Cacheable, isolated: true });
     expect(queryFnMock).toHaveBeenCalled();
   });
 
   test('fires mutation requests immediately without delay', () => {
-    spectator.service.registerHandler(buildMutationHandler());
+    spectator.service.registerHandler(buildMutationHandlerOne());
     const mutateFnMock = mockMutateMethod();
-    spectator.service.mutate(buildRequest());
+    spectator.service.mutate(buildRequestOne());
     expect(mutateFnMock).toHaveBeenCalled();
   });
 
-  test('collects and fires debounced quries together', fakeAsync(() => {
-    spectator.service.registerHandler(buildQueryHandler());
+  test('collects and fires different requests separately. Apollo bataches them', fakeAsync(() => {
+    spectator.service.registerHandler(buildQueryHandlerOne());
+    spectator.service.registerHandler(buildQueryHandlerTwo());
+    const queryFnMock = mockQueryMethod();
+    spectator.service.query(buildRequestOne()).subscribe();
+    spectator.service.query(buildRequestTwo()).subscribe();
+    tick(debounceTimeMs);
```

Comment:
should verify `queryFnMock` not called before tick, otherwise a non-batched implementation would pass here.